pkg/provider/provider.go

```diff
@@ -52,13 +46,247 @@ func Provider() *schema.Provider {
 			},
+			// todo: add database and schema once unqualified identifiers are supported
+			"warehouse": {
+				Type:        schema.TypeString,
+				Description: "Specifies the virtual warehouse to use by default for queries, loading, etc. in the client session. Can be sourced from SNOWFLAKE_WAREHOUSE environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_WAREHOUSE", nil),
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Description: "Specifies the role to use by default for accessing Snowflake objects in the client session. Can be sourced from the `SNOWFLAKE_ROLE` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_ROLE", nil),
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Description: "Snowflake region, such as \"eu-central-1\", with this parameter. However, since this parameter is deprecated, it is best to specify the region as part of the account parameter. For details, see the description of the account parameter. [Snowflake region](https://docs.snowflake.com/en/user-guide/intro-regions.html) to use.  Required if using the [legacy format for the `account` identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#format-2-legacy-account-locator-in-a-region) in the form of `<cloud_region_id>.<cloud>`. Can be sourced from the `SNOWFLAKE_REGION` environment variable.",
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_REGION", "us-west-2"),
+				Optional:    true,
+				Deprecated:  "Specify the region as part of the account parameter",
+			},
+			"validate_default_parameters": {
+				Type:        schema.TypeBool,
+				Description: "ValidateDefaultParameters disable the validation checks for Database, Schema, Warehouse and Role at the time a connection is established",
+				Optional:    true,
+			},
+			"params": {
+				Type:        schema.TypeMap,
+				Description: "Sets other connection (i.e. session) parameters. [Parameters](https://docs.snowflake.com/en/sql-reference/parameters)",
+				Optional:    true,
+			},
+			"session_params": {
+				Type:        schema.TypeMap,
+				Description: "Sets session parameters. [Parameters](https://docs.snowflake.com/en/sql-reference/parameters)",
+				Optional:    true,
+				Deprecated:  "Use `params` instead",
+			},
+			"client_ip": {
+				Type:        schema.TypeString,
+				Description: "IP address for network check",
+				Optional:    true,
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Description: "Either http or https. Can be sourced from `SNOWFLAKE_PROTOCOL` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PROTOCOL", "https"),
+			},
+			"host": {
+				Type:        schema.TypeString,
+				Description: "Supports passing in a custom host value to the snowflake go driver for use with privatelink.",
```

Comment:
Can you add in every description that it can be retrieved from other source e.g. here env (sometimes it's described and like in this case, it's not). From user's perspective it's pretty important to know how to setup the client without reading the code to know what's going on

Reply:
done